tests/type_inference/test_literals.py

```diff
@@ -125,6 +125,31 @@ def test_set_env():
     for value in global_values: assert isinstance(value, TypeVar)
 
 
+def test_set_name_unassigned():
+    """Test visitor for name nodes representing a single unassigned variable in module."""
+    program = "george"
+    module = _parse_text(program)
+    name_nodes = [n for n in module.nodes_of_class(astroid.Name)]
+    for name_node in name_nodes:
+        name_type_var = name_node.frame().type_environment.lookup_in_env(name_node.name)
+        assert name_node.type_constraints.type == name_type_var
+
+
+@given(cs.random_dict_variable_value(min_size=1))
+def test_set_name_assigned(variables_dict):
+    """Test visitor for name nodes representing a variables with assigned values in module."""
+    program = ""
+    # parse dictionary into input program
+    for variable_name in variables_dict:
+        assume(not iskeyword(variable_name))
+        program += variable_name + " = " + repr(variables_dict[variable_name]) + "\n" + variable_name + "\n"
```

Comment:
There is not for this one, the difference is that the helper has a random number of single-line-single-assignment statements, whereas in this test case I want the program to have the assignments as well a single line afterwards with just the variable name in order to get the actual Name nodes in the AST once parsed.